src/modules/schedule/controller.ts

```diff
@@ -0,0 +1,115 @@
+import { formatSchedule } from "./formatter";
+import { Schedule } from "../../generated-types/graphql";
+import { ScheduleModel } from "./model";
+import { ObjectID } from "bson";
+import { isBoolean } from "lodash";
+import { minimumViableSchedule, partialSchedule } from "./partial-schedules";
+
+
+// get the schedules for a user
+export async function getSchedulesByUser(userID: string): Promise<Schedule[]> {
+  const userSchedules = await ScheduleModel.find({created_by: userID})
+  if (userSchedules.length == 0) {
+    throw new Error("No schedules found for this user")
+  }
+  return userSchedules.map(formatSchedule)
+}
+
+// get the schedule for a user and a specific term
+export async function getScheduleByID(id: string): Promise<Schedule> {
+  const scheduleFromID = await ScheduleModel.findById({_id: id})
+  if (!scheduleFromID) {
+    throw new Error("No schedules found with this ID")
+  }
+  return formatSchedule(scheduleFromID as any)
+}
+
+// delete a schedule specified by ObjectID
+export async function removeSchedule(scheduleID: string): Promise<string> {
+  const deletedSchedule = await ScheduleModel.findByIdAndDelete(scheduleID)
+  if (!deletedSchedule) {
+    throw new Error("Schedule deletion failed")
+  }
+  return scheduleID
+}
+
+
+// create a new schedule
+export async function createSchedule(created_by: string, term: string, schedule_name: string | undefined | null, is_public: boolean, class_IDs: string[] | undefined | null, primary_section_IDs: string[], secondary_section_IDs: string[]): Promise<Schedule> {
+  
+  const newID = new ObjectID()
+
+  const schedulePartsToCreate: minimumViableSchedule = { _id: newID, created_by: created_by, term: term }
+  if (schedule_name) {
+    schedulePartsToCreate.name = schedule_name
+  }
+
+  if (class_IDs) {
+    schedulePartsToCreate.class_IDs = class_IDs
+  }
+
+  if (primary_section_IDs) {
+    schedulePartsToCreate.primary_section_IDs = primary_section_IDs
+  }
+
+  if (secondary_section_IDs) {
+    schedulePartsToCreate.secondary_section_IDs = secondary_section_IDs
+  }
+
+  schedulePartsToCreate.public = is_public
+
+  const newSchedule = await ScheduleModel.create(schedulePartsToCreate)
+  // const newSchedule = await ScheduleModel.create({_id: newID, name: schedule_name, created_by: created_by, date_created: current_time, last_updated: current_time, term: term, public: is_public, class_IDs: class_IDs, section_IDs: section_IDs})
```

Comment:
I think this would be much more readable if it had ternary operators instead of if statements. Something like 

```ts
{
  name: schedule_name ? schedule_name : undefined,
  class_IDs: class_IDs ? class_IDs : undefined,
}
```